inc/ThirdParty/Hostings/Wpengine.php

```diff
@@ -0,0 +1,220 @@
+<?php
+
+namespace WP_Rocket\ThirdParty\Hostings;
+
+use WP_Rocket\Event_Management\Subscriber_Interface;
+
+/**
+ * Compatibility class for wpengine
+ *
+ * @since 3.6.1
+ */
+class Wpengine implements Subscriber_Interface {
+	/**
+	 * Array of events this subscriber wants to listen to.
+	 *
+	 * @since 3.6.1
+	 *
+	 * @return array
+	 */
+	public static function get_subscribed_events() {
+		if (
+			! (
+				class_exists( 'WpeCommon' )
+				&&
+				function_exists( 'wpe_param' )
+			)
+		) {
+			return [];
+		}
+
+		return [
+			'rocket_varnish_field_settings'           => 'varnish_field',
+			'rocket_display_input_varnish_auto_purge' => 'return_false',
+			// Prevent mandatory cookies on hosting with server cache.
+			'rocket_cache_mandatory_cookies'          => [ 'return_empty_array', PHP_INT_MAX ],
+			'rocket_advanced_cache_file',
+			'return_empty_string',
+			'admin_init'                              => [
+				'remove_notices',
+				'run_rocket_bot_after_wpengine',
+			],
+			/**
+			 * Always keep WP_CACHE constant to true.
+			 *
+			 * @since 2.8.6
+			 */
+			'set_rocket_wp_cache_define'              => 'return_true',
+			/**
+			 * Conflict with WP Engine caching system.
+			 *
+			 * @since 2.6.4
+			 */
+			'do_rocket_generate_caching_files'        => 'return_false',
+			'after_rocket_clean_domain',
+			'clean_wpengine',
+			'rocket_buffer'                           => [ 'add_footprint', 50 ],
+			'rocket_disable_htaccess',
+			'disable_htaccess',
+			'wp_rocket_upgrade'                       => [ 'reset_htaccess', 11, 2 ],
+		];
```

Comment:
Syntax doesn't look correct here:
- The key should be the hook you want to add one or more callback(s) on
- The value should be either the callback, an array with the callback + the priority and number of parameters, or an array of array with the same format

You also don't need to keep the comments in there